relay-protocol/src/lib.rs

```diff
@@ -17,19 +17,23 @@
 )]
 
 mod annotated;
+mod condition;
 mod impls;
 mod macros;
 mod meta;
 mod size;
 mod traits;
+mod utils;
 mod value;
 
 pub use self::annotated::*;
+pub use self::condition::*;
```

Comment:
This module contains a lot of types that usually do not need to be imported. By glob-exporting them, we're littering the global crate namespace as can be seen in [the docs](https://getsentry.github.io/relay/relay_protocol/index.html) now.

When you compare the imports with how `relay-sampling` was structured, there was an explicit re-export of the `RuleCondition` type, as this is the most commonly used one, but all the other types were left in a `pub mod condition` instead. I'd prefer if we do this here, too.

That is independent of whether other modules are glob-exported or not.